src/parser/collect-stories.js

```diff
@@ -0,0 +1,138 @@
+/* eslint-env browser */
+import { logger } from '@storybook/client-logger';
+import { combineParameters } from '@storybook/client-api';
+import { extractId } from './extract-id';
+
+import RegisterContext from '../components/RegisterContext.svelte';
+import RenderContext from '../components/RenderContext.svelte';
+
+/* Called from a webpack loader and a jest transformation.
+ *
+ * It mounts a Stories component in a context which disables
+ * the rendering of every <Story/> and <Template/> but instead
+ * collects names and properties.
+ *
+ * For every discovered <Story/>, it creates a storyFn which
+ * instantiate the main Stories component: Every Story but
+ * the one selected is disabled.
+ */
+
+const createFragment = document.createDocumentFragment
+  ? () => document.createDocumentFragment()
+  : () => document.createElement('div');
+
+export default (StoriesComponent, stories) => {
+  const repositories = {
+    meta: null,
+    stories: [],
+  };
+
+  // extract all stories
+  try {
+    const context = new RegisterContext({
+      target: createFragment(),
+      props: {
+        Stories: StoriesComponent,
+        repositories,
+      },
+    });
+    context.$destroy();
+  } catch (e) {
+    logger.error(`Error extracting stories ${e.toString()}`, e);
+  }
+
+  const { meta } = repositories;
+  if (!meta) {
+    logger.error('Missing <Meta/> tag');
+    return {};
+  }
+
+  const { component: globalComponent } = meta;
+
+  // collect templates id
+  const templatesId = repositories.stories
+    .filter((story) => story.isTemplate)
+    .map((story) => story.id);
+
+  // check for duplicate templates
+  const duplicateTemplatesId = templatesId.filter(
+    (item, index) => templatesId.indexOf(item) !== index
+  );
+
+  if (duplicateTemplatesId.length > 0) {
+    logger.warn(
+      `Found duplicates templates id for stories '${meta.name}': ${duplicateTemplatesId}`
+    );
+  }
+
+  return {
+    meta,
+    stories: repositories.stories
+      .filter((story) => !story.isTemplate)
+      .reduce((all, story) => {
+        const { id, name, template, component, source = false, ...props } = story;
+
+        const storyId = extractId(story);
+        if (!storyId) {
+          return all;
+        }
+
+        const unknowTemplate = template != null && templatesId.indexOf(template) < 0;
```

Comment:
```suggestion
        const unknownTemplate = template != null && templatesId.indexOf(template) < 0;
```
(and references below)